MappingThread.py

```diff
@@ -117,19 +124,50 @@ def run(self):
                 if latitude == last_latitude and longitude == last_longitude:
                     continue
 
-                # Prevent update spam from freezing UI
+                # Prevent update spam
                 current_time = time.time()
                 if current_time - last_update_time < 0.5:
                     continue
 
-                self.plotMap(latitude, longitude)
-
-                # notify UI that new data is available to be displayed
-                self.sig_received.emit()
+                if self.plotMap(latitude, longitude):
+                    # notify UI that new data is available to be displayed
+                    self.sig_received.emit()
+                else:
+                    continue
 
                 last_latitude = latitude
                 last_longitude = longitude
                 last_update_time = current_time
 
             except Exception as ex:
                 print("Error in map thread loop: %s" % ex)
+
+
+# To be run in a new process as the stitching and resizing is a CPU bound task
+def processMap(requestQueue, resultQueue):
+    while True:
+        try:
+            (p1, p2, zoom, desiredSize) = requestQueue.get()
+
+            location = MapBox.TileGrid(p1, p2, zoom)
+            location.downloadArrayImages()
+
+            largeMapImage = location.genStichedMap()
+
+            if desiredSize is not None:
+                # Scale "to fit", maintains map aspect ratio even if it differs from the desired dimensions aspect ratio
+                scaleFactor = min(desiredSize[0] / largeMapImage.shape[0], desiredSize[1] / largeMapImage.shape[1])
+            else:
+                scaleFactor = 1
+
+            scaleFactor = min(scaleFactor, 1)  # You shall not scale the map larger. Waste of memory.
```

Comment:
Might want to document this somewhere like confluence for future modifications? Or make '1' not a magic number -> aka a constant

Reply:
Hm, yeah I think I'll turn it into a constant and add a comment. Good point.

It is linear scaling though so the number 1 isn't _that_ magic.

Reply:
That's a fair point. Could go either way then